### ISSUE TITLE
TreeGrid on v23 autowidth broken

### ISSUE THREAD

**Issue by KaijuDeluxe** [vaadin/flow-components#3607]
### Description of the bug

After upgradeing to vaadin 24 from 14 and createing a Treegrid like

```java
treeGrid.addHierarchyColumn(ProductCategory::getCategoryName).setHeader("Kategorie").setSortable(false).setAutowidth(true);
```
the column is not resized automatically, even when i call  treeGrid.recalculateColumnWidths() when new Objects are added or the grid is set with items. I must to setWidth("400px") to show the content

<img width="218" alt="image" src="https://user-images.githubusercontent.com/39457956/185729552-47081818-2a4d-4bc3-ae52-ccc71105cdb3.png">

EDIT: when i call in the dataprovide treeGrid.recalculateColumnWidths(); it works. But directly after creating the treegrid nothing happens, the columns are not resized. Only when i expand a node, the column get resized

### Expected behavior

The Column should be set autowidth like in vaadin 14

### Minimal reproducible example

```java
TreeGrid<ProductCategory> treeGrid = new TreeGrid<>();
treeGrid.addHierarchyColumn(ProductCategory::getCategoryName).setHeader("Kategorie").setSortable(false).setAutoWidth(true);

treeGrid.recalculateColumnWidths();
```

### Versions

- Vaadin / Flow version: 23
- Java version: 11
- OS version: Mac OS 12
- Browser version (if applicable): Safari and Chrome latests versions
- Application Server (if applicable):
- IDE (if applicable): IntelliJ

**Comment by web-padawan** [comment]
Need to check if this could be a regression from https://github.com/vaadin/web-components/pull/3525

**Comment by tomivirkki** [comment]
> Need to check if this could be a regression from https://github.com/vaadin/web-components/pull/3525

Seems like it. Turns out having `max-width: 100%` on the toggle element prevents grid from obtaining its intrinsic width while calculating auto-width for the column. We could probably fix this by using `display: flex` instead of `max-width: 100%` for the toggle.

A workaround for this issue is to apply the following styles to `vaadin-grid-tree-toggle`:
```
display: flex;
max-width: initial;
```

**Comment by jouni** [comment]
Have we considered `display: contents` for the `:host`?

**Comment by tomivirkki** [comment]
> Have we considered `display: contents` for the :host? Probably has its share of complications, but wondering if it could be viable here.

In tree-toggle's case, host defines the flexbox layout. If `display: contents` is used on the host, the layout is broken:
![Screenshot 2022-09-30 at 11 29 07](https://user-images.githubusercontent.com/1222264/193275988-866874a9-5676-42fa-9630-b9d036dce2f6.png)

> having max-width: 100% on the toggle element prevents grid from obtaining its intrinsic width while calculating auto-width for the column

After more investigation, turns out that using `max-width: 100%` doesn't actually affect calculating the column's intrinsic width. There is, however, some [CSS that applies a negative margin to the tree toggle element](https://github.com/vaadin/web-components/blob/14013efe21a3c1f911ff1f51b56b31024ecfb3db/packages/grid/theme/lumo/vaadin-grid-tree-toggle-styles.js#L13), probably added in an attempt to align the arrow icon with the column header title.

Without the negative margin:
![Screenshot 2022-09-30 at 15 58 19](https://user-images.githubusercontent.com/1222264/193276272-b075ffe4-ae22-4b81-9cef-58d484688059.png)

With the negative margin:
![Screenshot 2022-09-30 at 15 58 00](https://user-images.githubusercontent.com/1222264/193276344-a6896401-a25e-4ed0-8526-9117c897d897.png)

Using a negative margin does reduce the intrinsic width of the component and thus reduces the resulting width of an auto-width tree-column. Before applying `max-width: 100%`, the problem was still there, but it wasn't that obvious to see.

I suggest fixing the issue by using `transform: translateX` in place of the negative margin for shifting the component horizontally. Using a transform doesn't affect the component's width and thus fixes the automatically calculated tree column width.

**Comment by KaijuDeluxe** [comment]
awesome! In which version it will be rolled out?

**Comment by tomivirkki** [comment]
The fix should be included in the next patch releases of 23.1 and 23.2, also 23.3.0-(next alpha)
